data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/StartsWithExpressionFunction.java

```diff
@@ -0,0 +1,53 @@
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.model.event.Event;
+
+import javax.inject.Named;
+import java.util.List;
+import java.util.function.Function;
+
+@Named
+public class StartsWithExpressionFunction implements ExpressionFunction {
+    private static final int NUMBER_OF_ARGS = 2;
+
+    static final String STARTS_WITH_FUNCTION_NAME = "startsWith";
+    @Override
+    public String getFunctionName() {
+        return STARTS_WITH_FUNCTION_NAME;
+    }
+
+    @Override
+    public Object evaluate(
+            final List<Object> args,
+            final Event event,
+            final Function<Object, Object> convertLiteralType) {
+
+        if (args.size() != NUMBER_OF_ARGS) {
+            throw new RuntimeException("startsWith() takes exactly two arguments");
+        }
+
+        String[] strArgs = new String[NUMBER_OF_ARGS];
+        for (int i = 0; i < NUMBER_OF_ARGS; i++) {
+            Object arg = args.get(i);
+            if (!(arg instanceof String)) {
+                throw new RuntimeException(String.format("startsWith() takes only string type arguments. \"%s\" is not of type string", arg));
+            }
+            String str = (String) arg;
+            if (str.charAt(0) == '"') {
+                strArgs[i] = str.substring(1, str.length()-1);
+            } else if (str.charAt(0) == '/') {
+                Object obj = event.get(str, Object.class);
+                if (obj == null) {
+                    return false;
+                }
+                if (!(obj instanceof String)) {
+                    throw new RuntimeException(String.format("startsWith() takes only string type arguments. \"%s\" is not of type string", obj));
```

Comment:
```suggestion
                    throw new RuntimeException(String.format("startsWith() only operates on string types. The value at \"%s"\ is \"%s\" which is not a string type.", str, obj));
```

This error may be confusing. In this case, `startsWith()` is given a key and the value is not of type string.